subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/ecb.c

```diff
@@ -108,39 +211,49 @@ uint32_t ecb_encrypt_nonblocking(struct ecb *ecb)
 
 	/* start the encryption h/w */
 	nrf_ecb_task_trigger(NRF_ECB, NRF_ECB_TASK_STARTECB);
-
-	return 0;
 }
 
-static void ecb_cleanup(void)
+static void isr_ecb(const void *arg)
 {
-	/* stop h/w */
-	NRF_ECB->TASKS_STOPECB = 1;
-	nrf_ecb_task_trigger(NRF_ECB, NRF_ECB_TASK_STOPECB);
+	ARG_UNUSED(arg);
 
-	/* cleanup interrupt */
-	irq_disable(ECB_IRQn);
-}
+	if (NRF_ECB->EVENTS_ERRORECB) {
+#if defined(NRF54L_SERIES)
+		struct ecb *e = (void *)((uint8_t *)NRF_ECB->ECBDATAPTR -
+					 sizeof(struct ecb));
+#else /* !NRF54L_SERIES */
+		struct ecb *e = (void *)NRF_ECB->ECBDATAPTR;
+#endif /* !NRF54L_SERIES */
```

Comment:
Suggestion: Maybe it would be a bit nicer with this , the task stop and irq_disable above the if?

Reply:
yes, agree that its nicer to move the refer closer to the callback being invoked.